src/details/ArborX_DetailsLegacy.hpp

```diff
@@ -87,8 +87,29 @@ struct LegacyDefaultCallback
   }
 };
 
+struct LegacyDefaultTemplateValue
```

Comment:
Please briefly describe (either reply here or add to the PR description) why we introduce this tag rather than say values are indices + box.

Reply:
Index + box is a valid value, that can and will be used to construct (Basic)BVH. We want to introduce a type that will only be used to partially instantiate on, which must not be a valid type.

Reply:
I do not understand.

Reply:
I get it now.